learning_resources/filters_test.py

```diff
@@ -182,8 +182,12 @@ def test_learning_resource_filter_level():
     hs_resource = hs_run.learning_resource
     grad_resource = grad_run.learning_resource
 
-    hs_resource.runs.set([hs_run])
-    grad_resource.runs.set([grad_run])
```

Comment:
[RelatedManager.set](https://docs.djangoproject.com/en/5.0/ref/models/relations/#django.db.models.fields.related.RelatedManager.set) delegates adds new items to the relationship and removes excess items via the manager's `remove` method. But [`RelatedManager.remove`](https://docs.djangoproject.com/en/5.0/ref/models/relations/#django.db.models.fields.related.RelatedManager.remove) only exists for ManyToMany relationships and nullable ForeignKey relationships.

I.e., `.set(...)` only adds and does not remove for normal, non-nullable foreign key relationships.